mergify_engine/tests/unit/test_train.py

```diff
@@ -0,0 +1,210 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2020 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from unittest import mock
+
+import pytest
+
+from mergify_engine import context
+from mergify_engine import github_types
+from mergify_engine import merge_train
+from mergify_engine import utils
+
+
+@pytest.fixture()
+def redis():
+    r = utils.get_redis_for_cache()
+    r.flushdb()
+    try:
+        yield r
+    finally:
+        r.flushdb()
+
+
+def fake_train_car_create_pull(inner_self):
+    inner_self.tmp_pull_request_number = inner_self.pull_request_number + 10
+
+
+def fake_train_car_delete_pull(inner_self):
+    pass
+
+
+@pytest.fixture
+def monkepatched_traincar(monkeypatch):
+    monkeypatch.setattr(
+        "mergify_engine.merge_train.TrainCar.create_pull", fake_train_car_create_pull
+    )
+    monkeypatch.setattr(
+        "mergify_engine.merge_train.TrainCar.delete_pull", fake_train_car_delete_pull
+    )
+
+
+@pytest.fixture
+def fake_client():
+    client = mock.Mock()
+    client.get.return_value = mock.Mock(
+        json=mock.Mock(return_value={"commit": {"sha": "sha1"}})
+    )
+    return client
+
+
+def fake_pull(number, **kwargs):
```

Comment:
if you want to put that in utils it would be fine, I know I wanted it in a test already :)